streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -75,6 +83,26 @@ String segmentName(final long segmentId) {
     Segment getSegmentForTimestamp(final long timestamp) {
         return getSegment(segmentId(timestamp));
     }
+    
+    Segment getMaxSegmentLessThanEqualToTimestamp(final long timestamp) {
```

Comment:
Possibly don't need these. See comment above in `fetchAll`, but if we do then should add some tests

Reply:
+1. We can use the `segments` call to get the list of segments with the timestamp boundaries, and then rely on `SegmentIterator`.